fvm/src/gas/price_list.rs

```diff
@@ -609,21 +666,22 @@ impl PriceList {
     /// Returns the gas required for committing an object to the state blockstore.
     #[inline]
     pub fn on_block_link(&self, data_size: usize) -> GasCharge<'static> {
-        // TODO: The FIP makes it sound like this would need 2 memcpys, is that what's desired?
+        let size = data_size as i64;
+        let memcpy = self.block_memcpy_per_byte_cost.saturating_mul(size);
         GasCharge::new(
             "OnBlockLink",
-            self.block_link_base,
-            // data_size as i64 * self.block_link_per_byte_cost * self.storage_gas_multiplier,
-            self.block_link_per_byte_cost
+            self.block_link_base
+                .saturating_add((2 as i64).saturating_mul(memcpy)),
```

Comment:
Could we move the hard-coded `2` here to either price list or at least local constant?

Reply:
Can't do that because this is literally "twice the amount", but I will add a comment to clarify. Good catch.